### PR TITLE
[READY] Plumbing: People Milker!

### PR DESCRIPTION
**Pull request by Time-Green**
On the first day of christmas, Time-Green gave to me: People plumbing
image

Adds a new plumbing machine that lets you insert reagents and extract other reagents depending on species. Like you give a plant person NH3 or something and machine then extracts Light Heal Juice. 

Machine also functions as a plumbing forcefeeder, taking something out is optional. 

:cl:
add: You can now hook people into the plumbing people extractor to produce new rare reagents! 
/:cl:

I currently have a placeholder reaction that turns water into lemonade. It is not staying since I don't want it to be TOO weird. 
Planned reagents:

- [x] Acid to liquid gibs as default (scales with acid strength, 10u of sulp acid is 1 liquid gib and lots of pain)
- [x] Adds Essence of Drug. Created by feeding someone krokodile, methamphetamine, bath salts and protozine. Gives stun immunity, flight, glowing aura and pacifism. Overdosing will freeze the player and send them through walls towards the SM
- [x] Plant person creates light heal juice. 20u ammonia = 1u of this bad boy. Slow passive lighthealing (half of podpeople) with slow metabolism. On overdose, grass starts growing in your lungs and thats bad for you
- [x] Shadowperson and Nightmare give dark heal juice. No OD, 1u of liquid dark matter = 1u of the stuff. Absolutely no downsides to having this in your system for a long time
- [x] Lizard wine. 5 wine to 1 lizard wine. 
- [x] Organic Slurry > Fly person > nutrient
- [x] Bonehurting juice > skeleton > 5 milk. Harvest straight out of the bone
- [x] Bonehurting juice and stable plasma > plasmamen > ground plasma

Suggestions for more reactions are welcome!

Also this technically touches species so here:

image

### CODE REVIEW

**Comment by Misdoubtful** [comment]
"People milker" Jesus Christ.

**Comment by CRITAWAKETS** [comment]
force feeding people is going to have so many interesting possibilities

_infinite carrot juice intensifies_

**Comment by Time-Green** [comment]
> "People milker" Jesus Christ. Cool to have more use for the dead bodies piling up though.

I've already changed the name to be less suggestive. It's inherently weird, but I don't want it to be TOO weird

**Comment by Ghommie** [comment]
Take your time to rename the other paths, defines, procs and variables too. They are misleading; Milk has nothing to do with the feature.

**Comment by Time-Green** [comment]
> Take your time to rename the other paths, defines, procs and variables too. They are misleading; Milk has nothing to do with the feature.

Milking isn't exclusively from udders. You can milk for venom or milk almonds. It wasn't sexual until the people of /tg/ decided it was. I will rename the rest when I get the time

**Comment by Ghommie** [comment]
> 
> Milking isn't exclusively from udders. You can milk for venom or milk almonds.

I know, you already told it on discord too, but that doesn't justify how ambiguous it was.
Let's suppose there is this average Joe who is totally oblivious to what a "people milker" (now named "people extractor") is. One day he opens an otherwise empty text file and reads the two words, then he tries to guess what it's. I'm sure he'd be more inclined to guess it's something related to the thousands years old profession of the cowman or a milking machine, rather than some feeding-slash-colon-cleansing-apparatus inspired by a notorious horror movie. That's why I said the name (and references too, for coherence) needed disambiguation. Nothing risqué, nothing sexual.
Also sorry for not having argumented this before, I had supper.

**Comment by tralezab** [comment]
The OP has late stage gaming disorder

**Comment by Twaticus** [comment]
can moths give milkies

**Comment by ilikesaying** [comment]
can we harvest plasma from plasmamen with this

**Comment by TemporalOroboros** [comment]
Can this be used as an IV drip?

**Comment by IndieanaJones** [comment]
This is only a few steps away from Dwarf Fortress mermaid farms

**Comment by Time-Green** [comment]
> can we harvest plasma from plasmamen with this

> Can this be used as an IV drip?

It’s more of a trading system, not a blood giver. Although it can be used to intraveneously feed chemicals. I’ll have to make a blood drained machine later

**Comment by SkeletalElite** [comment]
Moth milk for /tg/ 2021

**Comment by Tlaltecuhtli** [comment]
can this be put the harvester machine that already exists  and do the same functionality? it ll look less of ............ people milker.................

**Comment by Pugie** [comment]
Make androids give nanomachines

**Comment by interestingusernam3** [comment]
>It wasn't sexual until the people of /tg/ decided it was.

Screenshot_2020-12-25 breast milking - e621

**Comment by Tlaltecuhtli** [comment]
can you make job specific reagents? like assistants giving quad sec for sec

**Comment by Time-Green** [comment]
> can you make job specific reagents? like assistants giving quad sec for sec

Totally. Adding recipys is a piece of cake. If you got a few good ideas, PM them to me or drop them here.

**Comment by Dorsisdwarf** [comment]
It's a pissmas bloody miracle.

File: code/datums/elements/plumbing_extractables/_plumbing_extractable.dm [hunk 1]
```diff
@@ -0,0 +1,45 @@
+///Element used for interfacing with the plumbing people extractor machine. Placed on the reagents datum
+/datum/element/plumbing_extractable
+	var/list/required_reagents = list(/datum/reagent/water = 2)
+	var/list/returned_reagents = list(/datum/reagent/consumable/lemonade = 1)
```
Comment (interestingusernam3) [inline code/datums/elements/plumbing_extractables/_plumbing_extractable.dm:4]: Jarate factory

**Comment by Pugie** [comment]
Mad-Max-Fury-Road

**Comment by Time-Green** [comment]
Right, there's 4 reactions in-game right now. 

-Acid to liquid gibs. The stronger the acid, the more liquid gibs. Obviously hurts 
-10u of krokodile, protozine, methamphetamine and bath salts gets you Essence of Drug. Gives flight, glowing aura and stun immunity. On overdose it slowly launches you through walls into the SM. Also has pacifism, I'll see about removing it if people don't end up making the thing in 10 minutes
-Lights Blessing. It's 20u of ammonia to 1u in a podperson. Heals really slow in light, slow metabolism. On OD grass grows in your lungs and slowly kills you
-Darkness' Blessing. 1u of liquid dark matter to 1u in a shadowperson or nightmare. No OD and has good healing in the darkness. Long term exposure eventually makes your skin dark, gives you nightvision, red eyes, dark skin and a weakness to light. 

Kinda out of creative juices for the moment. Still need to think of something for lizards, illuminescents, plasmamen and plasmamen. The rest of the species are an optional addition, and species can have multiple reactions. If anyone has some good idea's with recipys, please drop them here. Job specific is fine too

**Comment by Tlaltecuhtli** [comment]
banana juice -> clown -> clown tears

**Comment by ATH1909** [comment]
> > can you make job specific reagents? like assistants giving quad sec for sec
> 
> Totally. Adding recipys is a piece of cake. If you got a few good ideas, PM them to me or drop them here.

add a lemonade -> magical lemonade reaction for wizards as a reference to fairly odd parents

also add moth milk

**Comment by DrCrawler** [comment]
Add a coffee ---> hot chocolate reaction for humans.

**Comment by Dorsisdwarf** [comment]
> **Kinda out of creative juices for the moment. Still need to think of something for lizards, illuminescents, plasmamen, felinid and moth. The rest of the species are an optional addition, and species can have multiple reactions. If anyone has some good idea's with recipys, please drop them here. Job specific is fine too**

Lizard reaction should be extracting crocodile tears

**Comment by interestingusernam3** [comment]
>Add a coffee ---> hot chocolate reaction for humans. (1u coffee to 20u hot chocolate as it always happens)

only people without caffeine addictions will understand.

**Comment by Time-Green** [comment]
I'm gonna save the recipys for another PR and try to get this merged first. As usual, I'll update the wikipedia with the new machinery and recipys when it gets merged. For now the recipys are in the TODO list.

**Comment by interestingusernam3** [comment]
Can you make a plumbing cock-vorer next?

**Comment by optimumtact** [comment]
this is not a good way to produce base reagents

bad idea basically not even worth salvagaing

File: code/modules/reagents/chemistry/reagents/drink_reagents.dm [hunk 2]
```diff
@@ -215,6 +215,18 @@
 		holder.remove_reagent(/datum/reagent/consumable/capsaicin, 2)
 	..()
 
+/datum/reagent/consumable/milk/moth
```
Comment (Jared-Fogle) [inline code/modules/reagents/chemistry/reagents/drink_reagents.dm:218]: no

**Comment by wesoda25** [comment]
At first I was ambivalent but now this pr sucks

**Comment by ShibaInuLord** [comment]
Downvoting cause moth milk

File: code/modules/reagents/chemistry/reagents/drink_reagents.dm [hunk 3]
```diff
@@ -215,6 +215,18 @@
...
+	name = "Moth Milk"
+	description = "An opaque white liquid produced by moths."
+	color = "#DFDFDF" // rgb: 223, 223, 223
+	taste_description = "moth milk"
+	glass_icon_state = "glass_white"
+	glass_name = "glass of moth milk"
+	glass_desc = "White and nutritious goodness!"
+
+	taste_mult = DRINK_FANTASTIC
```
Comment (optimumtact) [inline code/modules/reagents/chemistry/reagents/drink_reagents.dm:227]: I agree

**Comment by Time-Green** [comment]
image
This is what could be

**Comment by Jared-Fogle** [comment]
sometimes you just have to take a stand on what the future of the game will be, and decide the people extractor isnt in it

**Comment by Time-Green** [comment]
rekt

**Comment by Jared-Fogle** [comment]
and i see that you closed this yourself so im going to double close it so that you cant reopen it

**Comment by Time-Green** [comment]
NO

**Comment by Time-Green** [comment]
cancel culture stikes again

**Comment by optimumtact** [comment]
Moth milkies

**Comment by Sparkezel** [comment]
Moth milkies

**Comment by interestingusernam3** [comment]
Moth milkies

**Comment by Jared-Fogle** [comment]
many are saying this!

**Comment by ExcessiveUseOfCobblestone** [comment]
Forced milking of "reagents" omfg

**Comment by Time-Green** [comment]
> Forced milking of "reagents" omfg

Feel free to reopen and merge it when you’re ready

**Comment by Low3** [comment]
Interesting idea, id suggest reworking it into corpse recycling, where you put corpse into a vat with tons acid and chems, where after few minutes it gets destroyed giving out resulting chems. It both saves the basic functionality and makes it gross in more accepltable way. But it destroys the body which in turn removes person from the round, so idk about that part.

**Comment by Time-Green** [comment]
> Interesting idea, id suggest reworking it into corpse recycling, where you put corpse into a vat with tons acid and chems, where after few minutes it gets destroyed giving out resulting chems. It both saves the basic functionality and makes it gross in more accepltable way. But it destroys the body which in turn removes person from the round, so idk about that part.

People dissolver? You might be onto something here...

**Comment by Low3** [comment]
Using bodies for getting powerfull chems is an interesting idea for traitors/morally questionable chemists. I would even suggest making it an uplink thing or tech locked so all of the those overpowered chems arent in the wild in the first 10 minutes.

File: code/modules/reagents/chemistry/reagents/drink_reagents.dm (hunk 3 above)
Comment (Stalkeros2) [reply code/modules/reagents/chemistry/reagents/drink_reagents.dm:227]: YEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEES

**Comment by Stalkeros2** [comment]
NO WAY
MOTH MILK
BASED AS **HELL**

**Comment by Stalkeros2** [comment]
Wait it's ancient af.(

**Comment by optimumtact** [comment]
do-you-ever-look-at-someone-and-wonder-what-is-going-inside-their-head

**Comment by Time-Green** [comment]
> Wait it's ancient af.(

It's a year old fuck you you're ancient

**Comment by ExcessiveUseOfCobblestone** [comment]
Readd the milker and call it Mommy

**Comment by SuperNovaa41** [comment]
2022 year of the people milker

**Comment by Kylerace** [comment]
cool pr

**Comment by Pickle-Coding** [comment]
I agree with opening it again, but I don't agree woth closing it again.

**Comment by Mothblocks** [comment]
this  pr was literally the turning point for me as a tg community member

**Comment by Time-Green** [comment]
> this pr was literally the turning point for me as a tg community member

And it can be again

**Comment by Kylerace** [comment]
it wouldnt let me press the green button sorry @Pickle-Coding

**Comment by TJatPBnJ** [comment]
its time

**Comment by EricZilla** [comment]
Lovers Lab ass PR

**Pull request closed without merging**